File: client/src/assets/gameService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/game';

const gameService = {
  async createNewGame(boardSize = 4) {
    const response = await axios.post(`${API_BASE_URL}/new?boardSize=${boardSize}`);
    return response.data;
  },

  async makeMove(gameId, direction) {
    const response = await axios.post(`${API_BASE_URL}/${gameId}/move?direction=${direction}`);
    return response.data;
  },

  async getGameState(gameId) {
    const response = await axios.get(`${API_BASE_URL}/${gameId}`);
    return response.data;
  },

  async restartGame(gameId, boardSize = 4) {
    const response = await axios.post(`${API_BASE_URL}/${gameId}/restart?boardSize=${boardSize}`);
    return response.data;
  }
};

export default gameService;

File: 2048-game/src/main/java/com/game2048/service/GameService.java
package com.game2048.service;

import com.game2048.model.Direction;
import com.game2048.model.GameState;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@Service
public class GameService {

    private final Map<String, GameState> games = new HashMap<>();

    public String createNewGame(int boardSize) {
        String gameId = UUID.randomUUID().toString();
        GameState initialState = GameState.initialize(boardSize);
        games.put(gameId, initialState);
        return gameId;
    }

    public GameState makeMove(String gameId, Direction direction) {
        GameState currentState = games.get(gameId);
        if (currentState == null) {
            throw new IllegalArgumentException("Game not found");
        }

        GameState newState = currentState.move(direction);
        games.put(gameId, newState);
        return newState;
    }

    public GameState getGameState(String gameId) {
        GameState state = games.get(gameId);
        if (state == null) {
            throw new IllegalArgumentException("Game not found");
        }
        return state;
    }

    public void restartGame(String gameId, int boardSize) {
        GameState newState = GameState.initialize(boardSize);
        games.put(gameId, newState);
    }

    public boolean gameExists(String gameId) {
        return games.containsKey(gameId);
    }
}
